indexer/src/main/java/au/org/aodn/esindexer/utils/AssociateRecordsUtil.java

```diff
@@ -0,0 +1,90 @@
+package au.org.aodn.esindexer.utils;
+
+import au.org.aodn.esindexer.model.MediaType;
+import au.org.aodn.esindexer.model.RelationType;
+import au.org.aodn.stac.model.LinkModel;
+import au.org.aodn.stac.util.JsonUtil;
+
+import java.util.*;
+
+import static au.org.aodn.esindexer.utils.CommonUtils.safeGet;
+
+public class AssociateRecordsUtil {
+
+    private record TitleWithAbstract(String recordTitle, String recordAbstract) {}
+
+    public static List<LinkModel> generateAssociatedRecords(Map<String, ?> data) {
+        var records = new ArrayList<LinkModel>();
+        if (data == null || data.isEmpty()) {
+            return records;
+        }
+        var parentsRecord = getParentRecord(data);
+        if (!parentsRecord.isEmpty()) {
```

Comment:
no "s" after "parent"

Reply:
I didn't check all every single line, but please double check adjective and plural usages